core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -451,6 +529,45 @@ private ExprValue exprDate(ExprValue exprValue) {
     }
   }
 
+  /**
+   * DateTime implementation for ExprValue.
+   *
+   * @param dateTime ExprValue of String type.
+   * @return ExprValue of date type.
+   */
+  private ExprValue exprDateTime(ExprValue dateTime, ExprValue timeZone) {
```

Comment:
Let's discuss this method. I'd like to understand a few things:
- How do we know `dateTime` can be successfully parsed? This is done in two places without try-catch, yet it was parsing was wrapped in try-catch elsewhere.
-   We are checking if timeZone is null but not dateTime -- how come?
-  If timeZone.isNull block is special case only for `exprDateTimeNoTZ` then it needs to move to that method. 
- try block on line 549 guards against parse on line 550 failing? Then why is it safe to assume parse call on line 558 will succeed?
- What is the difference between LocalDateTime and default time zone?

Reply:
> We are checking if timeZone is null but not dateTime -- how come?

The null field will come from `exprDateTimeNoTimezone` which is used when only one parameter to datetime is called.

If I call `SELECT DATETIME('2021-02-31 12:34:56)` then `exprDateTimeNoTimezone` calls exprDateTime with null in the timezone field.

Reply:
A better pattern would be to have function overloads: one with date and timezone, and one without timezone. 
As it stands, you should include the `null` as an expected/handled expected @param value for timeZone.